internal/prober/prober.go

```diff
@@ -26,37 +29,41 @@ import (
 
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	defaultGetSecretBackoff             = 100 * time.Millisecond
 	defaultGetSecretMaxAttempts         = 3
 	backOffDurationForThrottledRequests = 10 * time.Second
+	// expiryBufferFraction is used to compute a revised expiry time used by the prober to determine expired leases
+	// Using a fraction allows the prober to intervene before KCM marks a node as unknown, but at the same time allowing
+	// kubelet sufficient retries to renew the node lease.
+	// Eg:- nodeLeaseDuration = 40s, kcmNodeMonitorGraceDuration = 40s, kubeletRenewTime = 10s.
+	// 		The node lease will be considered expired by the prober at 30s. This allows kubelet 3 attempts instead of 4
+	// 		to renew the node lease.
+	expiryBufferFraction = 0.75
+	nodeLeaseNamespace   = "kube-node-lease"
 )
 
 // Prober represents a probe to the Kube ApiServer of a shoot
 type Prober struct {
-	namespace           string
-	config              *papi.Config
-	client              client.Client
-	scaler              dwdScaler.Scaler
-	shootClientCreator  ShootClientCreator
-	internalProbeStatus probeStatus
-	externalProbeStatus probeStatus
-	ctx                 context.Context
-	cancelFn            context.CancelFunc
-	l                   logr.Logger
+	namespace          string
+	config             *papi.Config
+	scaler             dwdScaler.Scaler
+	shootClientCreator ShootClientCreator
```

Comment:
In shootclient.go can you following camel casing for name of `shootclientCreator` struct. It should be renamed to `shootClientCreator` instead.